politeiawww/cmsuser_test.go

```diff
@@ -0,0 +1,442 @@
+			usernameFresh,
+			pwd,
+			"123456",
+			www.UserError{
+				ErrorCode: www.ErrorStatusVerificationTokenInvalid,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error wrong verification",
+			emailFresh,
+			usernameFresh,
+			pwd,
+			"this is an invalid verification token",
+			www.UserError{
+				ErrorCode: www.ErrorStatusVerificationTokenInvalid,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error duplicate pubkey",
+			emailFresh,
+			usernameFresh,
+			pwd,
+			replyFresh.VerificationToken,
+			www.UserError{
+				ErrorCode: www.ErrorStatusDuplicatePublicKey,
+			},
+			hex.EncodeToString(id.Public.Key[:]),
+		},
+		{
+			"error invalid pubkey",
+			emailFresh,
+			usernameFresh,
+			pwd,
+			replyFresh.VerificationToken,
+			www.UserError{
+				ErrorCode: www.ErrorStatusInvalidPublicKey,
+			},
+			"this is a bad pubkey",
+		},
+		{
+			"error duplicate username",
+			emailFresh,
+			username,
+			pwd,
+			"12345",
+			www.UserError{
+				ErrorCode: www.ErrorStatusDuplicateUsername,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error malformed username too short",
+			emailFresh,
+			usernameTooShort,
+			pwd,
+			"123456",
+			www.UserError{
+				ErrorCode: www.ErrorStatusMalformedUsername,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error malformed username too long",
+			emailFresh,
+			usernameTooLong,
+			pwd,
+			"123456",
+			www.UserError{
+				ErrorCode: www.ErrorStatusMalformedUsername,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error malformed username reg exp",
+			emailFresh,
+			usernameRegExp,
+			pwd,
+			"123456",
+			www.UserError{
+				ErrorCode: www.ErrorStatusMalformedUsername,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error malformed password too short",
+			emailFresh,
+			usernameFresh,
+			passwordTooShort,
+			"123456",
+			www.UserError{
+				ErrorCode: www.ErrorStatusMalformedPassword,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"success fresh",
+			emailFresh,
+			usernameFresh,
+			pwd,
+			replyFresh.VerificationToken,
+			nil,
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+		{
+			"error user not found",
+			"notfound@example.org",
+			"notfound",
+			pwd,
+			"123456",
+			www.UserError{
+				ErrorCode: www.ErrorStatusVerificationTokenInvalid,
+			},
+			hex.EncodeToString(idFresh.Public.Key[:]),
+		},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			registerReq := cms.RegisterUser{
+				Email:             v.email,
+				Username:          v.username,
+				Password:          v.pwd,
+				VerificationToken: v.token,
+				PublicKey:         v.pubkey,
+			}
+			_, err = p.processRegisterUser(registerReq)
+			got := errToStr(err)
+			want := errToStr(v.wantError)
+			if got != want {
+				t.Errorf("got error %v, want %v", got, want)
+			}
+		})
+	}
+}
+
+func TestTempInviteUser(t *testing.T) {
```

Comment:
Can you put these tests in the `TestInviteNewUser` function?  Keeping them setup as a seperate set of tests is fine.  We want them all all to be in one place though so that if someone goes looking for test coverage of a certain function they know it will all be in `TestFunctionName` and they won't have to look around.

Reply:
done